smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/connectors/WorkerPoolRegistry.java

```diff
@@ -105,39 +101,45 @@ private WorkerExecutor getWorker(String workerName) {
                 .forEach(m -> defineWorker(m.getJavaMember()));
     }
 
-    private void defineWorker(Method method) {
+    public void defineWorker(String className, String method, String poolName) {
+        Objects.requireNonNull(className, "className was empty");
         Objects.requireNonNull(method, "Method was empty");
 
-        Blocking blocking = method.getAnnotation(Blocking.class);
-
-        // Validate @Blocking is used in conjunction with @Incoming, or @Outgoing
-        if (!(method.isAnnotationPresent(Incoming.class) || method.isAnnotationPresent(Outgoing.class))) {
-            throw getBlockingError(method, "no @Incoming or @Outgoing present");
-        }
-
-        if (!blocking.value().equals(Blocking.DEFAULT_WORKER_POOL)) {
+        if (!poolName.equals(Blocking.DEFAULT_WORKER_POOL)) {
             // Validate @Blocking value is not empty, if set
-            if (Validation.isBlank(blocking.value())) {
-                throw getBlockingError(method, "value is blank or null");
+            if (Validation.isBlank(poolName)) {
+                throw getBlockingError(className, method, "value is blank or null");
             }
 
             // Validate @Blocking worker pool has configuration to define concurrency
-            String workerConfigKey = WORKER_CONFIG_PREFIX + "." + blocking.value() + "." + WORKER_CONCURRENCY;
-            Optional<Integer> concurrency = configInstance.get().getOptionalValue(workerConfigKey, Integer.class);
+            String workerConfigKey = WORKER_CONFIG_PREFIX + "." + poolName + "." + WORKER_CONCURRENCY;
+            Optional<Integer> concurrency = ConfigProvider.getConfig().getOptionalValue(workerConfigKey, Integer.class);
             if (!concurrency.isPresent()) {
-                throw getBlockingError(method, workerConfigKey + " was not defined");
+                throw getBlockingError(className, method, workerConfigKey + " was not defined");
             }
 
-            workerConcurrency.put(blocking.value(), concurrency.get());
+            workerConcurrency.put(poolName, concurrency.get());
         }
     }
 
-    private IllegalArgumentException getBlockingError(Method method, String message) {
-        return new IllegalArgumentException(
-                "Invalid method annotated with @Blocking: " + methodAsString(method) + " - " + message);
+    private void defineWorker(Method method) {
+        Objects.requireNonNull(method, "Method was empty");
+
+        Blocking blocking = method.getAnnotation(Blocking.class);
+
+        String methodName = method.getName();
+        String className = method.getDeclaringClass().getName();
+
+        // Validate @Blocking is used in conjunction with @Incoming, or @Outgoing
```

Comment:
Should this validation be in the other method so it doesn't need to be repeated in a runtime?

Not sure what that means for parameters though

Reply:
I don't know. This check was done this way before my changes too.

Reply:
Maybe pass an array of annotations on the method?

Reply:
Not sure what you mean. Quarkus doesn't do this check?

My presumption was that Quarkus is calling the defineWorker version that doesn't have the check. If it's calling the one with the check, that's fine

Reply:
oh, sorry, I misunderstood your question :)
It seems I missed it and Quarkus does not have this validation. I created an issue for it, will fix it tomorrow: https://github.com/quarkusio/quarkus/issues/8567

For Quarkus we don't have info on the present annotations here, it's taken from Jandex so has to be done before calling `defineWorker`

Reply:
@kenfinnigan I checked Quarkus once again and there `@Blocking` on methods that don't have `@Incoming` or `@Outgoing` will be ignored. That's because the logic for blocking is applied only to methods from `MediatorBuildItem`s.

Anyway, such validation makes sense and I will add it to Quarkus.